packages/jupyter-ai/jupyter_ai/config_manager.py

```diff
@@ -146,14 +149,17 @@ def _init_validator(self) -> Validator:
             self.validator = Validator(schema)
 
     def _init_config(self):
+        default_config = self._init_defaults()
         if os.path.exists(self.config_path):
-            self._process_existing_config()
+            self._process_existing_config(default_config)
         else:
-            self._create_default_config()
+            self._create_default_config(default_config)
 
-    def _process_existing_config(self):
+    def _process_existing_config(self, default_config):
         with open(self.config_path, encoding="utf-8") as f:
-            config = GlobalConfig(**json.loads(f.read()))
+            existing_config = json.loads(f.read())
+            merged_config = {**existing_config, **default_config}
```

Comment:
Does this give precedence to the `existing_config` or would override values from `default_config`? Also, this is doing a shallow merge of properties, which might not be always desirable.

Reply:
This will override values from `default_config`.